o2a/mappers/spark_mapper.py

```diff
@@ -47,60 +47,63 @@ def __init__(
-        self.archive_extractor = ArchiveExtractor(oozie_node=oozie_node, params=self.params)
-        self.prepare_command = None
+        self.file_extractor = FileExtractor(oozie_node=oozie_node, property_set=self.property_set)
+        self.archive_extractor = ArchiveExtractor(oozie_node=oozie_node, property_set=self.property_set)
         self.hdfs_files: List[str] = []
         self.hdfs_archives: List[str] = []
         self.dataproc_jars: List[str] = []
 
     def on_parse_node(self):
-
-        if self.has_prepare:
-            self.prepare_command = self.get_prepare_command(oozie_node=self.oozie_node, params=self.params)
-
         _, self.hdfs_files = self.file_extractor.parse_node()
         _, self.hdfs_archives = self.archive_extractor.parse_node()
 
-        self.java_jar = self._get_or_default(self.oozie_node, SPARK_TAG_JAR, None, params=self.params)
-        self.java_class = self._get_or_default(self.oozie_node, SPARK_TAG_CLASS, None, params=self.params)
+        self.java_jar = self._get_or_default(self.oozie_node, tag=SPARK_TAG_JAR)
+        self.java_class = self._get_or_default(self.oozie_node, tag=SPARK_TAG_CLASS)
         if self.java_class and self.java_jar:
             self.dataproc_jars = [self.java_jar]
             self.java_jar = None
-        self.job_name = self._get_or_default(self.oozie_node, SPARK_TAG_JOB_NAME, None, params=self.params)
+        self.job_name = self._get_or_default(self.oozie_node, tag=SPARK_TAG_JOB_NAME)
 
         job_xml_nodes = xml_utils.find_nodes_by_tag(self.oozie_node, SPARK_TAG_JOB_XML)
 
         for xml_file in job_xml_nodes:
             tree = ET.parse(source=xml_file.text)
-            self.properties.update(self._parse_config_node(tree.getroot()))
+            self.job_properties.update(self._parse_config_node(tree.getroot()))
 
         config_nodes = xml_utils.find_nodes_by_tag(self.oozie_node, SPARK_TAG_CONFIGURATION)
         if config_nodes:
-            self.properties.update(self._parse_config_node(config_nodes[0]))
+            self.job_properties.update(self._parse_config_node(config_nodes[0]))
 
         spark_opts = xml_utils.find_nodes_by_tag(self.oozie_node, SPARK_TAG_OPTS)
         if spark_opts:
-            self.properties.update(self._parse_spark_opts(spark_opts[0]))
+            self.job_properties.update(self._parse_spark_opts(spark_opts[0]))
 
         app_args = xml_utils.find_nodes_by_tag(self.oozie_node, SPARK_TAG_ARGS)
         for arg in app_args:
-            self.application_args.append(el_utils.replace_el_with_var(arg.text, self.params, quote=False))
+            self.application_args.append(
+                el_utils.replace_el_with_var(arg.text, self.property_set, quote=False)
+            )
 
-    @staticmethod
-    def _get_or_default(root: ET.Element, tag: str, default: str = None, params: Dict[str, str] = None):
+    def _get_or_default(self, root: ET.Element, tag: str, default: str = None):
```

Comment:
This method looks very similar to `get_tag_el_text`?

Reply:
True! Fixed in e8cd0f510d38afcd96da054d70f4fd61037f20d0